### PR TITLE
Feature/api for receiving menu information

### LINKED ISSUE #184: API for receiving menu information
**User Story:**

As a developer I want an API to retrieve menus that are stored in the database so I can display this information on the menu pop-ups.

**Acceptance Criteria:**

* Can successfully call API and receive menu to be displayed

**Additional Context:**

* Menus are stored in the database as images (jpg)
* Blocked by #183 which sets up database to support menus and provides dummy data

### PR DESCRIPTION
**Pull request by HongShi10**
## Link to Issue Number
Closes #184 

## Description
Added routes for the menus, tests we need to be added in another issue for these routes.

## Checklist
- [x] All feature request A/C have been met OR the bug has been fixed
- [ ] I ran all necessary tests and they pass
- [x] I rebased upstream/master onto my working branch before opening this PR
- [x] I have named my PR something sensible
- [x] I have included the issue number above
- [x] I have assigned at least two people to review my changes

## Other Notes
Had a bit of troubling with rebasing so the commits may be duplicated. However, it should all be working fine.

### CODE REVIEW

**Comment by HongShi10** [comment]
Also added tags for swagger for easier navigation

File: backend/routes/reservation.js [hunk 1]
```diff
@@ -89,7 +87,8 @@ router.get('/', async (req, res) => {
  *
  * /reservation/restaurant:
  *   get:
- *     description: Fetch all reservations for a restaurant
+ *     tags: [Reservation]
+ *     description: Fetch a all reservation for a restaurant
```
Comment (wryao64) [inline backend/routes/reservation.js:91]: I think you overrode changes for the description, and for the POST and PUT endpoints too. Can you please fix this?

File: backend/tests/reservation/getAllForRestaurant.js [hunk 2]
```diff
@@ -41,19 +39,19 @@ describe('GET reservations/', () => {
...
-    //   .asyncQuery(
-    //     `INSERT INTO RESERVATION (ID, Date, Time, Notes, NumberOfGuests, TableID, RestaurantID, UserID)
-    // VALUES (1, '2020-03-14', '11:30:00', '', '1', '1', '1', '1');`
-    //   )
-    //   .then(({ error }) => {
-    //     assert.isUndefined(error, 'Expected no errors when adding items to database');
-    //   });
+    await connection
+      .asyncQuery(
+        `INSERT INTO restaurant_db.RESERVATION (ID, Date, Time, Notes, NumberOfGuests, TableID, RestaurantID, UserID)
```
Comment (wryao64) [inline backend/tests/reservation/getAllForRestaurant.js:49]: I think 'restaurant_db.' was removed so different database names could be used. Could you please check and remove these from the other test cases too (other test files as well)? Thanks

**Comment by HongShi10** [comment]
Closing PR to redo as too many errors

**Pull request closed without merging**